kratos/processes/calculate_discontinuous_distance_to_skin_process.cpp

```diff
@@ -36,19 +37,35 @@ namespace Kratos
 		: mFindIntersectedObjectsProcess(rVolumePart, rSkinPart)
 		, mrSkinPart(rSkinPart)
 		, mrVolumePart(rVolumePart)
-		, mOptions(CalculateDiscontinuousDistanceToSkinProcessFlags::CALCULATE_ELEMENTAL_EDGE_DISTANCES.AsFalse())
+		, mOptionEdge(CalculateDiscontinuousDistanceToSkinProcessFlags::CALCULATE_ELEMENTAL_EDGE_DISTANCES.AsFalse())
+		, mOptionExtraEdge(CalculateDiscontinuousDistanceToSkinProcessFlags::CALCULATE_ELEMENTAL_EXTRAPOLATED_EDGE_DISTANCES.AsFalse())
 	{
 	}
 
 	template<std::size_t TDim>
 	CalculateDiscontinuousDistanceToSkinProcess<TDim>::CalculateDiscontinuousDistanceToSkinProcess(
 		ModelPart& rVolumePart,
 		ModelPart& rSkinPart,
-		const Flags rOptions)
+		const Flags rOptionEdge)
 		: mFindIntersectedObjectsProcess(rVolumePart, rSkinPart)
 		, mrSkinPart(rSkinPart)
 		, mrVolumePart(rVolumePart)
-		, mOptions(rOptions)
+		, mOptionEdge(rOptionEdge)
+		, mOptionExtraEdge(CalculateDiscontinuousDistanceToSkinProcessFlags::CALCULATE_ELEMENTAL_EXTRAPOLATED_EDGE_DISTANCES.AsFalse())
```

Comment:
I think this is wrong. If the user sets the `CALCULATE_ELEMENTAL_EXTRAPOLATED_EDGE_DISTANCES` flag to true you are changing its value.

Reply:
This would be the constructor for the user to only give the `CALCULATE_ELEMENTAL_EDGE_DISTANCES` and not gove the `CALCULATE_ELEMENTAL_EXTRAPOLATED_EDGE_DISTANCES` at all

Reply:
The point is that with one flag variable you can store 64 values by concatenating them. The value of `mOptions` is the result of a bitwise operation, meaning that we can combine both `CALCULATE_ELEMENTAL_EDGE_DISTANCES` and `CALCULATE_ELEMENTAL_EXTRAPOLATED_EDGE_DISTANCES`in one flag. Then when asking for the value it will check the corresponding bit (0 and 1 in this case, as there is only 2 local flags). Hope this explains. Otherwise we can discuss it in detail privately.

Reply:
And flags that are given in `rOptions` are `True` automatically, and local flags which are not given in `rOptions` are `False` automatically?